src/bloqade/squin/rewrite/qasm2/qasm2_modified_to_squin.py

```diff
@@ -81,3 +78,21 @@ def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
         node.replace_by(invoke_stmt)
 
         return RewriteResult(has_done_something=True)
+
+    def rewrite_glob_UGate(self, node: glob.UGate) -> RewriteResult:
+
+        # assume that QASM2DirectToSquin has already run, which converts QRegNew to qubit.new in the squin dialect.
+        # Then you're left with an ilist of ilists of qubit.news, for which a new squin.broadcast.u3 should be created.
+
+        ilist_of_registers = node.registers.owner
+        assert isinstance(ilist_of_registers, ilist.New)
+
+        for register_ilist in ilist_of_registers.values:
```

Comment:
If we want to be really fancy, we could do this with an `ilist_of_registers.map(squin.broadcast.u3)` or so, but this is also fine.